### ISSUE TITLE
Support jvmTarget 17

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#4287]
Currently setting it to 17 results in `Invalid value passed to --jvm-target`

**Comment by eygraber** [comment]
It seems like the issue is that I tried doing this in an included build. It works fine otherwise. Not sure why it doesn't work in included builds, but I'm guessing it's not a detekt issue.

**Comment by eygraber** [comment]
This project has a `buildSrc` that works fine with `jvmTarget = "17"` but the included build doesn't.

I just ran into this issue in another project that tries to use `jvmTarget = "17"` for a regular module, but it doesn't work (`Invalid value passed to --jvm-target`).

Maybe there is something going on here?

**Comment by eygraber** [comment]
Here's my config:

```
detekt {
  toolVersion = libs.versions.detekt.get()

  source.from("build.gradle.kts")

  autoCorrect = true
  parallel = true

  buildUponDefaultConfig = true

  config = project.files("${project.rootDir}/detekt.yml")
}

tasks.withType<Detekt>().configureEach {
  // Target version of the generated JVM bytecode. It is used for type resolution.
  jvmTarget = libs.versions.jdk.get()
}
```

**Comment by chao2zhang** [comment]
Kotlin 1.5.3X seems to support up to Java 16 as the target: https://github.com/JetBrains/kotlin/blob/1.5.30/compiler/config.jvm/src/org/jetbrains/kotlin/config/JvmTarget.kt.

I see that JVM_17 was added in Kotlin 1.6.0, so hopefully this is unblocked by https://github.com/detekt/detekt/pull/4277

**Comment by 3flex** [comment]
> I see that JVM_17 was added in Kotlin 1.6.0, so hopefully this is unblocked by #4277

I've manually tested, and yes, this should fix the issue. This issue will be closed when that PR is merged.

**Comment by vladimirfx** [comment]
Working workaround:

```kotlin
dependencies {
    detekt("io.gitlab.arturbosch.detekt:detekt-cli:1.19.0")
    detekt("org.jetbrains.kotlin:kotlin-compiler-embeddable:1.6.0")
}
```

**Comment by eygraber** [comment]
This is still happening sporadically in 1.20.0-RC2

Maybe related to detekt getting set up in a Gradle precompiled script plugin, which uses Kotlin 1.5.31

**Comment by cortinico** [comment]
> Looks like the issue is that the snapshots aren't updated since November.

Not sure what you refer to, but the last snapshot was published today:
https://oss.sonatype.org/content/repositories/snapshots/io/gitlab/arturbosch/detekt/

**Comment by eygraber** [comment]
Ah I was looking at a different view of the snapshot repo.

Well then this is still an issue. Like I said above:

> Maybe related to detekt getting set up in a Gradle precompiled script plugin, which uses Kotlin 1.5.31

**Comment by 3flex** [comment]
Do you have a complete reproducer to share? It shouldn't matter that detekt is setup in a precompiled script plugin - the classpath used by detekt when it runs should be isolated from that used by Gradle. We can verify with a build scan (that would also help in the absence of a complete repro).

**Comment by eygraber** [comment]
I'm working on putting something together. It also could be that I have custom rules that are on an older version of detekt, so I'm going to try to update those first.

**Comment by eygraber** [comment]
False alarm, it was only happening when I used `main-SNAPSHOT`, and it looks like the issue is that I set `toolVersion` to the same version that I use for the dependency, and it didn't like using `main-SNAPSHOT` for that. So I changed `toolVersion` to use `1.20.0-RC2` and then it worked.

It's a little odd because I made a fresh project with all of the same configuration and it didn't have this issue. Maybe something weird got stuck in the cache.
